deps/rcheevos/src/rcheevos/condition.c

```diff
@@ -107,8 +106,13 @@ rc_condition_t* rc_parse_condition(const char** memaddr, rc_parse_state_t* parse
     return 0;
   }
 
-  self->oper = rc_parse_operator(&aux);
+  result = rc_parse_operator(&aux);
+  if (result < 0) {
+    parse->offset = result;
+    return 0;
+  }
 
+  self->oper = (char)result;
```

Comment:
this is the change here. capture and handle the negative value as an `int` before storing it in the `char`.